docs/mathematics/number-power-of-2.mdx

```diff
@@ -0,0 +1,162 @@
+---
+title: Number is Power of 2
+slug: /mathematics/number-power-of-2
+---
+
+## Introduction
+
+Some Numbers can be represented as power of 2. Let us say `64` = `2^6`.
+But some numbers can not be represented as power of 2. Such as `22`, `45` and so on.
+
+
+## Explanation
+
+Let us have one example:
+
+```
+Let us say, number  = 256
+Number can be expressed as, 256 = 2^8
+Clearly we see, the number 256 is power of 2.
+But let's take another number, say number = 567.
+Since , it is an odd number.
+Clearly we see, the number 567 is not power of 2.
+```
+
+## Algorithm
+
+ - Given a number `number` to check whether it can represented as power of `2` or not.
+ - Well, if the number is power of `2`, then it's binary representation will contain only `1` setbit.
+ - for example `16` = `10000` in binary representation. 
+ - after subtracting 1 from the number all unset bits will become set and set bits will be unset.
+ - for example `15` = `01111` in binary representation.
+ - this two numbers bitwise AND `&` will be zero 
+ - `10000 & (1000 - 1)` -> `0000`
+ - if the number is not Power of `2` , then their bitwise AND will not zero. 
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+bool number_is_power_of_2(int number)
+{
+  /* if bitwise and of number and number - 1 is zero
+    then we can say that number is power of 2
+    otherwise it is not.
+    */
+    if((number & (number - 1)) == 0)
+        return true;
+    return false;
+}
+int main()
+{
+    cout << "Enter the number \n";
+    int number;
+    cin >> number;
+    bool solve = number_is_power_of_2(number);
+    if(solve)
+        cout << "The Number is power of 2\n";
+    else
+        cout << "No. the number is not power of 2\n";
+}
+
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
+import java.util.*;
+import java.lang.*;
+public class Powerof2 
+{
+	
+	// This function will tell us whether a number is power of 2
+	static boolean number_is_power_of_2(int number) 
+	{
+		/* if bitwise and of number and number-1 is zero
```

Comment:
Nitpicking here, but if you can align these comment blocks properly, would look good.